backend/app/core/security.py

```
@@ -197,6 +198,20 @@ async def get_current_user_from_session(request: Request) -> Dict:
+            # so both requests resolve to the one shared record instead of erroring.
+            logger.info(
+                f"Concurrent auto-create race for {user_id}; using the existing record"
+            )
+            from app.db.user import get_user_by_auth_id
+
+            user = await get_user_by_auth_id(user_id)
+            if user is None:
+                raise HTTPException(
+                    status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+                    detail="Failed to create user account. Please try signing in again or contact support."
+                )
```

Ruff (0.15.20)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)